website/content/blog/2023-11-reduce-reuse-rebase-buildpacks.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "Reduce, Reuse, Rebase: Sustainable Containers with Buildpacks"
+description: Buildpacks are the greenest choice for container builds
+slug: 2023-11-reduce-reuse-rebase-buildpacks
+weight: 1
+date:   2023-11-17 00:00:00 +0000
+author: Joe Kutner
+---
+
+Container builds can be very wasteful. Every operating system update, new dependency version, and toolchain upgrade results in copious amounts of energy used to build and rebuild our container images; often unnecessarily. It can be costly at scale, which is why Cloud Native Buildpacks were designed to perform builds as efficiently as possible.
```

Comment:
Thoughts
* Maybe add one sentence at the start to get the reader into the mindset. (ofc we expect that the reader knows docker etc.)
* `Container builds can be very wasteful.` does not actually say much. The next sentence says more.
* `which is why Cloud Native Buildpacks were designed to perform builds as efficiently as possible.` this is quite long and does not really say anything. Every tool wants so be performant. What comes in the next paragraphs rebasing etc. are interesting because these are the concepts that differ.. 
* To me, a history perspective is missing here. Some details into docker are described a bit later, but I sense that there is something missing here. (feel free to disagree, ofc.)
* Perhaps you want to merge both paragraphs. I am not sure why to break at this point.

Reply:
How about this?

>  which is why Cloud Native Buildpacks were designed to perform full builds only when a rebuild is actually required.

Re: history perspective, I worry that might clutter the intro. But maybe it is necessary to set context. Are you thinking I should add some of the history of buildpacks (ex. they pre-date `Dockerfile`). Also, it may be worth clarifying that the choice is not between Docker and Buildpacks, but rather `Dockerfile` and Buildpacks (buildpacks image work with docker and otehr container runtimes).

For the intro sentence, I'm not sure what you mean by "get the reader into the mindset". Can you elaborate?

Reply:
> How about this?
> [...]

👍 

> I worry that might clutter the intro. But maybe it is necessary to set context

Yeah, the history part should come later, but for the flow and to get into the context it would consider adding / teasing it in the intro. 

> Also, it may be worth clarifying that the choice is not between Docker and Buildpacks, but rather Dockerfile and Buildpack

Yes! I would take this in. You could think about using `Containerfile` instead of `Dockerfile`. 

> For the intro sentence, I'm not sure what you mean by "get the reader into the mindset"

Essentially, my comment is about setting the context. The blogs we have are very different in terms of content. Just to make it easier to change context. Buildpacks is used for XYZ.